state-sync/inter-component/event-notifications/src/lib.rs

```diff
@@ -0,0 +1,458 @@
+            .storage
+            .read()
+            .reader
+            .deref()
+            .batch_fetch_resources_by_version(config_access_paths, version)
+            .map_err(|error| {
+                Error::UnexpectedErrorEncountered(format!(
+                    "Failed to batch fetch resources at version: {:?}. Error: {:?}",
+                    version, error
+                ))
+            })?;
+
+        // Fetch the account state blob
+        let (account_state_blob, _) = self
+            .storage
+            .read()
+            .reader
+            .get_account_state_with_proof_by_version(config_address(), version)
+            .map_err(|error| {
+                Error::UnexpectedErrorEncountered(format!(
+                    "Failed to fetch account state with proof {:?}",
+                    error
+                ))
+            })?;
+        let account_state_blob = account_state_blob.ok_or_else(|| {
+            Error::UnexpectedErrorEncountered("Missing account state blob!".into())
+        })?;
+
+        // Fetch the new epoch from storage
+        let epoch = AccountState::try_from(&account_state_blob)
+            .and_then(|state| {
+                Ok(state
+                    .get_configuration_resource()?
+                    .ok_or_else(|| {
+                        Error::UnexpectedErrorEncountered(
+                            "Configuration resource does not exist!".into(),
+                        )
+                    })?
+                    .epoch())
+            })
+            .map_err(|error| {
+                Error::UnexpectedErrorEncountered(format!(
+                    "Failed to fetch configuration resource! Error: {:?}",
+                    error
+                ))
+            })?;
+
+        // Return the config payload
+        Ok(OnChainConfigPayload::new(
+            epoch,
+            Arc::new(
+                ON_CHAIN_CONFIG_REGISTRY
+                    .iter()
+                    .cloned()
+                    .zip_eq(on_chain_configs)
+                    .collect(),
+            ),
+        ))
+    }
+}
+
+#[async_trait]
+impl EventNotificationSender for EventSubscriptionService {
+    async fn notify_events(
+        &mut self,
+        version: Version,
+        events: Vec<ContractEvent>,
+    ) -> Result<(), Error> {
+        if events.is_empty() {
+            return Ok(()); // No events!
+        }
+
+        // Notify event subscribers and check if a reconfiguration event was processed
+        let reconfig_event_processed = self.notify_event_subscribers(version, events)?;
+
+        // If a reconfiguration event was found, also notify the reconfig subscribers
+        // of the new configuration values.
+        if reconfig_event_processed {
+            self.force_notify_reconfigurations(version).await
+        } else {
+            Ok(())
+        }
+    }
+
+    async fn force_notify_reconfigurations(&mut self, version: Version) -> Result<(), Error> {
```

Comment:
Maybe `notify_from_storage` or `reconcile_from_storage`, this is about getting the latest config from storage in the event there's no event?  `Force` implies you're doing it on purpose as a manual or purposeful action, but it looks like you use it above whenever you get a `reconfig` event.

Reply:
Yeah, the use of `force` was to say this is a manual invocation. The reason I call it in `notify_events` is just to re-use code 😄. Tbh, I'm more inclined to keep it as it is because I don't want the `EventSubscriptionService` to use the term `storage` in its API at all. IMO, storage is an internal concept (whether or not the service uses storage is irrelevant to the subscribers). But I feel like this is a little subjective :)

Reply:
I don't know it's still feels kinda weird, considering the one is called `notify_events` and this one is called `force_notify_reconfigurations`.  Maybe just `notify_reconfigurations`?  It isn't always manual, and usually you'll have a `force_...` if there's also a non-forcing function.

Reply:
As synced offline, we'll go with `notify_initial_configs` 😄